kodila-patterns2/src/main/java/com/kodilla/patterns2/adapter/bookclassifier/libraryb/Statistics.java

```diff
@@ -0,0 +1,41 @@
+package com.kodilla.patterns2.adapter.bookclassifier.libraryb;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public class Statistics implements BookStatistics {
+    @Override
+    public int averagePublicationYear(Map<BookSignature, Book> books) {
+        if (books.size() == 0) {
+            return 0;
+        }
+
+        int sum = 0;
+        for (Map.Entry<BookSignature, Book> entry : books.entrySet()) {
+            sum += entry.getValue().getYearOfPublication();
+        }
+
+        return sum / books.size();
+    }
+
+    @Override
+    public int medianPublicationYear(Map<BookSignature, Book> books) {
+        if (books.size() == 0) {
+            return 0;
+        }
+
+        int[] years = new int[books.size()];
+        int n = 0;
+        for (Map.Entry<BookSignature, Book> entry : books.entrySet()) {
+            years[n] = entry.getValue().getYearOfPublication();
+            n++;
+        }
+
+        Arrays.sort(years);
+        if (years.length % 2 == 0) {
+            return years[(int) (years.length / 2 + 0.5)];
```

Comment:
tu coś dziwnego robisz :) Operator "/" dla liczb typu int zwraca int. Czyli do inta dodajesz 0,5 i rzutujesz na inta, więc masz to, co mialeś. Jaki tego sens?